components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -231,6 +238,56 @@ public boolean doPreSetUserClaimValuesWithID(String userID, Map<String, String>
         return true;
     }
 
+    /**
+     * Validate whether the claim update request is from a provisioned user.
+     *
+     * @param userID           User id.
+     * @param profileName      Profile name.
+     * @param userStoreManager Userstore manager.
+     * @throws UserStoreException if an error occurred while retrieving the user claim list.
+     */
+    private void validateProvisionedUserClaimUpdate(String userID, String profileName,
+                                                    UserStoreManager userStoreManager) throws UserStoreException {
+
+        /*
+        Check whether the config is enabled to edit synced attributes of the provisioned user.
+        If it is enabled, blocking the attribute editing is not required.
+         */
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(SYNCED_ATTRIBUTE_EDITING_ENABLED))) {
+            return;
+        }
+
+        /*
+        Check whether this is an attribute syncing flow by checking the PROVISIONED_USER thread local property.
+        If it is an attribute syncing flow, blocking the attribute editing is not required.
+         */
+        if (IdentityUtil.threadLocalProperties.get().get(PROVISIONED_USER) != null &&
+                (Boolean) IdentityUtil.threadLocalProperties.get().get(PROVISIONED_USER)) {
+            return;
+        }
+
+        String username = ((UniqueIDJDBCUserStoreManager) userStoreManager).getUserNameFromUserID(userID);
+        Claim[] userClaimValues = userStoreManager.getUserClaimValues(username, profileName);
+
+        /*
+        If the PROVISIONED_SOURCE_CLAIM is already set as claim of the user, that user can be identified as provisioned
+        user. Therefore attribute editing will be blocked for that user.
+         */
+        for (Claim claim : userClaimValues) {
+            String claimURI = claim.getClaimUri();
+            if (!PROVISIONED_SOURCE_CLAIM.equals(claimURI)) {
+                continue;
+            }
+            if (StringUtils.isNotBlank(claim.getValue())) {
+                String errorMsg = "The user: " + username + " has been provisioned from federated IDP: " +
+                        claim.getValue() + ". Hence provisioned user attributes are not allowed to update";
+                throw new UserStoreClientException(errorMsg);
```

Comment:
Any chance we can throw an error code?

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/367/commits/9d70788d485e475ac471678bec53d31f24aca382